api/v1beta1/disruption_webhook.go

```diff
@@ -170,3 +189,135 @@ func (r *Disruption) getMetricsTags() []string {
 
 	return tags
 }
+
+// initialSafetynets runs the initial safety nets for any new disruption
+// returns true if any safety net were caught and returns any errors when attempting to run the safety nets
+func (r *Disruption) initialSafetynets() ([]string, error) {
+	responses := []string{}
+	// handle initial safety nets if safemode is enabled
+	if r.Spec.Unsafemode == nil || !r.Spec.Unsafemode.DisableAll {
+		if caught, err := safetyNetCountNotTooLarge(*r); err != nil {
+			return nil, fmt.Errorf("error checking for countNotTooLarge safetynet: %w", err)
+		} else if caught {
+			logger.Debugw("the specified count represents a large percentage of targets in either the namespace or the kubernetes cluster", r.Name, "SafetyNet Catch", "Generic")
+
+			responses = append(responses, "the specified count represents a large percentage of targets in either the namespace or the kubernetes cluster")
+		}
+
+		if r.Spec.Network != nil {
+			if caught := safetyNetNeitherHostNorPort(*r); caught {
+				logger.Debugw("The specified disruption either contains no Hosts or contains a Host which has neither a port or a host. The more ambiguous, the larger the blast radius.", r.Name, "SafetyNet Catch", "Network")
+
+				responses = append(responses, "The specified disruption either contains no Hosts or contains a Host which has neither a port or a host. The more ambiguous, the larger the blast radius.")
+			}
+		}
+	}
+
+	return responses, nil
+}
+
+// safetyNetCountNotTooLarge is the safety net regarding the count of targets
+// it will check against the number of targets being targeted and the number of targets in the k8s system
+// > 66% of the k8s system being targeted warrants a safety check if we assume each of our targets are replicated
+// at least twice. > 80% in a namespace also warrants a safety check as namespaces may be shared between services.
+// returning true indicates the safety net caught something
+func safetyNetCountNotTooLarge(r Disruption) (bool, error) {
+	if r.Spec.Unsafemode != nil && r.Spec.Unsafemode.DisableCountTooLarge {
+		return false, nil
+	}
+
+	userCount := r.Spec.Count
+	totalCount := 0
+	namespaceCount := 0
+	namespaceThreshold := 0.8
+	clusterThreshold := 0.66
+
+	if r.Spec.Unsafemode.Config != nil && r.Spec.Unsafemode.Config.CountTooLarge != nil {
+		if r.Spec.Unsafemode.Config.CountTooLarge.NamespaceThreshold != 0 {
```

Comment:
It seems there is no way to disable the namespace threshold without disabling the cluster threshold. Are we okay with that? I might be fine with a disruption that targets 100% of a namespace, but not if it's also targeting 80% of the cluster.

Reply:
if the user sets up the namespace targeting threshold at 100% and the cluster threshold as 80% they can accomplish what you're asking. Let me know if I am misunderstanding.

Reply:
Oh, great point, I missed that we were doing a `>` check instead of a `>=` check.